bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gmos/InstGmosCommon.java

```diff
@@ -423,9 +431,22 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
      * (Update5: SCI-0107)
      */
     public double getSetupTime(ISPObservation obs) {
-        if (_disperser.isMirror()) return 6 * 60;
-        if (_fpu.isSpectroscopic() || _fpu.isNSslit()) return 16 * 60;
-        return 18 * 60.;
+        if (_disperser.isMirror()) return SETUP_TIME_IMAGING;
+        if (_fpu.isSpectroscopic() || _fpu.isNSslit()) return SETUP_TIME_LS_SPECTROSCOPY;
+        return SETUP_TIME_IFU_MOS;
+    }
+
+    public double getSetupTime(Config conf) {
+        String fpu_name = (String) conf.getItemValue(FPU_KEY);
```

Comment:
I'm surprised this is a `String`.  Or "shocked but not surprised" or whatever they say these days.  Are we sure it isn't either a `GmosSouthType.FPUnitSouth` or `GmosNorthType.GPUnitNorth`?  Hopefully it is because then we can just directly ask whether it is IFU, etc. I would expect this method to be implemented something like:

```java
import java.util.Optional;
...
public double getSetupTime(Config conf) {
    return Optional.ofNullable(conf.getItemValue(FPU_KEY))
                   .map(c -> (GmosCommonType.FPUnit) c)
                   .filter(f -> !f.isImaging())
                   .map(f -> f.isIFU() ? SETUP_TIME_IFU_MOS : SETUP_TIME_LS_SPECTROSCOPY)
                   .orElse(SETUP_TIME_IMAGING);
 }
```

Reply:
Indeed... Thanks for pointing this out, I will fix some other stray Strings as well.

Reply:
Whether the value in the `Config` is a `String` or an `FPUnit` will depend on how the sequence was created.  It appears that [this happens in `ItcParametersProvider`](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/ItcParametersProvider.scala#L82-L84).  If I'm correct then the use of the `IDENTITY_MAP` there means that it will be an `FPUnit`.

Reply:
Actually, since people are in a hurry to test this feature, I will stop looking for things to improve (there are too many of them), and add just the changes you requested.